### PR TITLE
Improve type mapping documentation for Kafka connector

### PR DESCRIPTION
**Pull request by tlblessing**
## Description

Add a type mapping section. Relo SQL Support to bottom of topic for consistency.

Original PR that I had to close: https://github.com/trinodb/trino/pull/14147

### CODE REVIEW

**Review by jhlodin — APPROVED** [review]

File: docs/src/main/sphinx/connector/kafka.rst [hunk 1]
```diff
@@ -517,9 +501,22 @@ used as the target for a message. If a message includes a key, the producer will
...
+.. _kafka-type-mapping:
+
+Type mapping
 ------------
 
+Because Trino and Kafka each support types that the other does not, this
+connector :ref:`maps some types <type-mapping-overview>` when reading
+(:ref:`decoding <kafka-row-decoding>`) or writing (:ref:`encoding
+<kafka-row-encoding>`) data. Type mapping depends on the format (Avro, raw,
+JSON, CSV).
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/kafka.rst:513]: Reorder as Raw, Avro, JSON, CSV for compatibility with subsequent paragraph.

File: docs/src/main/sphinx/connector/kafka.rst [hunk 2]
```diff
@@ -1136,17 +1145,18 @@ which can be specified via ``dataFormat`` attribute.
 | |                                   | ``seconds-since-epoch``                                                        |
 +-------------------------------------+--------------------------------------------------------------------------------+
 
+No other types are supported.
 
 Default field decoder
-^^^^^^^^^^^^^^^^^^^^^
+"""""""""""""""""""""
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/kafka.rst:1151]: Could you change `"` to `+`? This is part of json decoder.

File: docs/src/main/sphinx/connector/kafka.rst [hunk 3]
```diff
@@ -1164,7 +1174,7 @@ For ``TIMESTAMP WITH TIME ZONE`` and ``TIME WITH TIME ZONE`` data types, if time
 be used as Trino value. Otherwise result time zone will be set to ``UTC``.
 
 ``avro`` decoder
-^^^^^^^^^^^^^^^^
+""""""""""""""""
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/kafka.rst:1177]: Could you use same naming between decoder and encoder? 
Screen Shot 2022-10-21 at 8 54 25

Comment (tlblessing) [reply docs/src/main/sphinx/connector/kafka.rst]: image

Comment (tlblessing) [reply docs/src/main/sphinx/connector/kafka.rst]: @ebyhr  thank you for your review

**Review by jhlodin — APPROVED** [review]
LGTM % any further comments from Yuya

**Review by ebyhr — APPROVED** [review]

**Comment by ebyhr** [comment]
Merged, thanks!

**Pull request merged**